lib/client/interfaces.go

```diff
@@ -36,8 +37,35 @@ import (
 	"golang.org/x/crypto/ssh/agent"
 )
 
+// KeyIndex helps to identity a key in the store.
+type KeyIndex struct {
+	// ProxyHost is the root proxy hostname that a key is associated with.
+	ProxyHost string
+	// Username is the username that a key is associated with.
+	Username string
+	// ClusterName is the cluster name that a key is associated with.
+	ClusterName string
+}
+
+// CheckAllSet verifies the KeyIndex is fully specified.
+func (idx KeyIndex) CheckAllSet() error {
```

Comment:
nit: rename to `Check`
most of our types use either `Check` or `CheckAndSetDefaults` for self-validation